### PR TITLE
Fix mergify for docs PRs

### PR DESCRIPTION
**Pull request by CriesofCarrots**
#### Problem
Mergify seems broken on docs PRs (eg https://github.com/solana-labs/solana/pull/29017/checks?check_run_id=9830706173), presumably since https://github.com/solana-labs/solana/pull/28852

#### Summary of Changes
Fix
It might be enough to use just the `check-skipped=error_reporting` condition. Wdyt @yihau ?

### CODE REVIEW

**Comment by CriesofCarrots** [comment]
Side note: it's super weird that mergify wants `build & deploy` instead of `build_and_deploy`

File: .mergify.yml [hunk 1]
```diff
@@ -41,8 +41,10 @@ pull_request_rules:
         - author≠@dont-squash-my-commits
         - or:
           # only require docs checks if docs files changed
-          - status-success=docs-build
           - -files~=^docs/
+          - and:
+            - status-success=build & deploy
+            - check-skipped=error_reporting
```
Comment (CriesofCarrots) [inline .mergify.yml:47]: Do you know if there is a way to specify the workflow of the github-action job? I didn't see one.
If not, we might need to make job-names more specific, as I see we're adding another `error_reporting` in #29009

**Comment by yihau** [comment]
ah! According to their docs: `GitHub Actions works slightly differently. To match a status check when using GitHub Action, only the job name is used.`

I guess it takes `build & deploy` cuz I assign it as the job name here.
https://github.com/solana-labs/solana/blob/a901fccb0064d3cc5f8bf6e6404a0543bb0f70ab/.github/workflows/docs.yml#L72

File: .mergify.yml (hunk 1 above)
Comment (yihau) [reply .mergify.yml:47]: I think the job name is the only thing they can recognize.

btw, is it okay to remove `- check-skipped=error_reporting` 🤔 
the error_reporting only works when any previous steps failed and I think we can ignore its status.

Comment (CriesofCarrots) [reply .mergify.yml:47]: I *think* that we can adjust to only one criterion: `- status-success=build & deploy`
Any build that touches docs files must run `build & deploy`, right?

Also, how about we rename that job to "build & deploy docs" to be more explicit for this use?

Comment (CriesofCarrots) [reply .mergify.yml]: Okay, updated to the single criterion @yihau . (Needs rebase on #29072 )

Comment (yihau) [reply .mergify.yml]: just merged 😈

**Review by yihau — APPROVED** [review]
🚀

**Pull request merged**